### ISSUE TITLE
transformer_layer_indexed hardcodes the Q4K kernel for Q/K/attn_output under a comment claiming exhaustive dispatch (silent corruption, currently masked)

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#3822]
Found while diagnosing #3804. **Currently masked — see below — but it is a silent-corruption defect, and the mask is the thing that will disappear.**

## The defect

`crates/aprender-serve/src/cuda/executor/layers/transformer_layer_indexed.rs`:

```rust
// line 78-81
let q = self.q4k_gemv_indexed_async(layer_weights.attn_q_ptr, &normed, q_dim, hidden_dim)?;
let k = self.q4k_gemv_indexed_async(layer_weights.attn_k_ptr, &normed, kv_dim, hidden_dim)?;
// PMAT-232 CONTRACT: Exhaustive dispatch — no catch-all. See tensor-layout-v1.yaml quant_dispatch.
// NOTE: Indexed async path only has Q4K/Q6K kernels. Other types must use workspace path.
let v = match layer_weights.attn_v_qtype {
    WeightQuantType::Q6K => { … }
    WeightQuantType::Q4K => { … }
    WeightQuantType::Q5K | Q8_0 | Q4_0 | Q5_0 | Q4_1 => return Err(…),
};
```

and again at line 100:

```rust
// 4. Output projection using indexed pointer (no sync)
let projected = self.q4k_gemv_indexed_async(layer_weights.attn_output_ptr, &attn_out, hidden_dim, q_dim)?;
```

**Q, K and `attn_output` are dispatched to the Q4K kernel unconditionally, with no qtype check** — directly above and below a `match` that carries the comment *"PMAT-232 CONTRACT: Exhaustive dispatch — no catch-all"* and that correctly refuses every non-Q4K/Q6K type. The contract is enforced for V only; the other three projections ignore it.

This is the same class as #2753 / PERF-050, where `dispatch_mul_mat` passed `WeightQuantType::Q4K` unconditionally and the first Q6_K weight the forward reached was dequantized with the Q4_K kernel, producing absmax 4.9e8 with 47/256 non-finite values.

## Why nobody has hit it

It is unreachable on every model tried so far, because the V arm is checked in the same function and errors first for anything outside Q4K/Q6K, so the layer bails to the workspace path before the bad Q/K results are used. On `qwen2.5-coder-0.5b-instruct-q4_k_m`, `attn_v` is Q8_0 ×12 / Q5_0 ×12, so the V arm always errors and the hardcoded Q/K never ship a result.

**The mask is a coincidence of tensor mix, not a guard.** Any model whose `attn_v` is Q4K or Q6K while `attn_q`/`attn_k`/`attn_output` are not — a perfectly legal GGUF, and exactly the kind llama.cpp produces when only some rows are K-quant-eligible (see the #3804 composition table: `ffn_down` stayed Q6_K/Q4_K while every 896-wide row fell back to Q5_0) — passes the V guard and then silently decodes Q/K/O with the wrong kernel. That is garbage logits with nothing reporting an error, i.e. the PMAT-781/783 silent-garbage class.

## Ask

Dispatch Q, K and `attn_output` through the same qtype `match` as V (or through `gemv_dispatch`), so the "no catch-all" contract covers all four projections. A regression test with `attn_v = Q4K` and `attn_q/k/o = Q5_0` would turn this RED today.

Refs #3804, #2753.

**Comment by noahgift** [comment]
milestone → 0.71.0 per docs/roadmaps/milestone-triage-2026-09-23.md (#4024), operator approval 2026-09-23 (relayed by aprender-cf). Reason: transformer_layer_indexed hardcodes the Q4K kernel: every quant must work (#3994)
